### PR TITLE
Mshullick/presdkv2

### PR DESCRIPTION
**Pull request by emptonaut**

### CODE REVIEW

File: pyenv_setup.sh [hunk 1]
```diff
@@ -57,6 +58,7 @@ set_up_env() {
     echo "setting up env ${ENV_NAME} with Python ${PYTHON_VERSION}"
     pyenv install ${PYTHON_VERSION}
     pyenv virtualenv ${PYTHON_VERSION} ${ENV_NAME}
+	exit 1
```
Comment (etonmollusk) [inline pyenv_setup.sh]: why exit 1 here

File: pyenv_setup.sh [hunk 2]
```diff
@@ -51,7 +52,6 @@ set_up_env() {
     if [ $? -ne 0 ];
     then
 	echo "pyenv command not found!"
-	exit 1
```
Comment (etonmollusk) [inline pyenv_setup.sh:54]: i think we do want to exit here, or else line 58 and 59 would fail

File: app.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
+""" Customer's application for Alpha SDK.
+
+This file contains a method imported and run by the SDK. All that is required
```
Comment (emptonaut) [inline app.py]:
```suggestion
This file contains a function imported and run by the SDK. All that is required
```

File: config.toml.example [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+# Maximum amount of buffers in the queue for appsink in gstreamer pipeline
+# 1 by default, 0 == unlimited
+#appsink-max-buffers = 1
+
+[application]
+example-setting = "hello!"
+iterations = 5
+# This directory is assumed to exist, primarily for use in the container
+output-dir = "/app/data"
+#output-dir = "/home/ubuntu/shullick/friendly-fortnight/container_data"
```
Comment (emptonaut) [inline config.toml.example]: Remove

File: install_docker.sh
Comment (emptonaut) [inline install_docker.sh:1]: Rename file to `install_nvidia_docker.sh` and add check that we're running on a jetson

File: install_industrialnext_deps.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+set -e
+
+ENV_NAME=${ENV_NAME:py310}
+PYTHON_BIN=${PYTHON_BIN:=$HOME/.pyenv/versions/$ENV_NAME/bin/python}
+
+# Ensure we have the credentials available
+[ -n "$INDUSTRIALNEXT_PY_INDEX" ] || { echo "Error: 'INDUSTRIALNEXT_PY_INDEX' is not set. Please see the README.md" >&2; exit 1; }
```
Comment (emptonaut) [inline install_industrialnext_deps.sh]:
```suggestion
[ -n "$INDUSTRIALNEXT_PY_INDEX" ] || { echo "Error: 'INDUSTRIALNEXT_PY_INDEX' is not set. Please see the documentation" >&2; exit 1; }
```

File: pyenv_setup.sh (hunk 2 above)
Comment (emptonaut) [inline pyenv_setup.sh:54]: Put back

File: sdk.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+def parse_args():
+    """
+    Parse configuration based on CLI args.
+    Return config dict if we can, else None.
+    """
+    parser = argparse.ArgumentParser(
+        description="SDK 1.0 Alpha",
+    )
+    parser.add_argument(
```
Comment (emptonaut) [inline sdk.py:36]: Need to add a default or required flag

File: pyenv_setup.sh [hunk 7]
```diff
@@ -1,17 +1,19 @@
...
 install_pyenv() {
     # Check if git installed
     git --version 2>&1 >/dev/null
-    
+
     if [ $? -ne 0 ];
     then
         echo "Installing git..."
-        sudo apt install git
+         apt install -y git
```
Comment (etonmollusk) [inline pyenv_setup.sh]: one extra space looks like

File: sdk.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    )
+    args = parser.parse_args()
+    if args.print_camera_modes:
+        camlib.print_cybersight_modes()
+        return None
+    config = read_config(args.config)
+    return config
+
+
+def read_config(file_path: str):
```
Comment (emptonaut) [inline sdk.py]: type hint

File: sdk_docker.sh [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+RUN_CMD="time docker run -it --rm --runtime nvidia \\
+    -v /tmp/argus_socket:/tmp/argus_socket \\
+    -v %s:/app/config.toml \\
+    %s \\
+    presdk:latest -c /app/config.toml %s"
+
+print_usage () {
+    SUBCMD_HELP_FMT="%8s: %s\n"
+    echo "Helper script for running docker commands associated with SDK Alpha"
+    echo "Usage: $0 <build,run,print> [--print-camera-modes|-h]"
```
Comment (emptonaut) [inline sdk_docker.sh:36]: Add sdk flags to the end

File: sdk_docker.sh [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+    printf "%16s\n" "Default config file is $CONFIG_FILE_DEFAULT"
+    printf "$SUBCMD_HELP_FMT" "build" "Build the presdk container"
+    printf "$SUBCMD_HELP_FMT" "run" "Run the presdk container with the given config file"
+    printf "$SUBCMD_HELP_FMT" "print" "Just print the template commands and exit"
```
Comment (emptonaut) [inline sdk_docker.sh:40]: add buildrun

File: pyenv_setup.sh (hunk 7 above)
Comment (emptonaut) [reply pyenv_setup.sh]: putting back the sudo, maybe I was drunkski

**Review by etonmollusk — APPROVED** [review]

**Pull request merged**